keps/sig-storage/2923-csi-migration-ceph-rbd/README.md

```diff
@@ -0,0 +1,49 @@
+# In-tree Storage Plugin to CSI Migration - Ceph RBD Design Doc
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+  - [New Feature Gates](#new-feature-gates)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+
+## Summary
+
+This document present as a vendor specific KEP for the parent KEP
+[CSI Migration](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration)
+
+This inherits all the contents from its parent KEP. It will introduce two new feature gates to be 
+used as as described in its parent KEP. For all other contents, please refer to the parent KEP.
+
+### New Feature Gates
+
+- CSIMigrationCephRBD
+  - As describe in [CSI Migration](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration), 
+  when this feature flag && the `CSIMigration` is enabled at the same time, the in-tree volume 
+  plugin that the cloud-provider use will be redirect to use the corresponding CSI driver. From a 
+  user perspective, nothing will be noticed.
+- InTreePluginCephRBDUnregister
```

Comment:
This feature gate isn't present or described in https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration/README.md -- is there precedent for it? What happens if this feature flag is enabled but the others are not?

I think it's a bit confusing for end-users to have so many feature flags controlling an identical feature. Why is this controlled with a feature flag rather than a config?

I think you will need to ensure this is accounted for in the beta questionnaire.

Reply:
Each feature flag serves a different purpose:
- CSIMigration: The core feature that controls the vendor-neutral logic
- CSIMigrationVendor: Enables the feature for a specific vendor
- InTreePluginVendorUnregister: Disables the in-tree plugin from being registered with Kubernetes. This actually can be used independently from CSIMigration* flags. Is your suggestion that this particular flag may be better suited as a config? Given that this flag is still [alpha](https://github.com/kubernetes/kubernetes/commit/bda557b4bc3f5ec655d32a9ec23345ed9c77147b) we have the opportunity to change it, but I think I would actually not track this as part of the CSI migration feature. It's a more general problem of how do we disable in-tree volume plugins.

Reply:
Yes, as Michelle has already pointed out. The flags serves all different purpose and there is no way to replace one with another. And technically speaking, this `InTreePluginCephRBDUnregister` flag is not part of the CSI migration design. It is just happen to be related with CSI Migration feature. Itself is actually another big question which is in-tree volume plugins. I will rephrase this paragraph to reflect this. Thanks for the advice!

Reply:
I try to rephrase and make it more clear. Please help to take a look! Thanks and appreciate!

Reply:
I also make another PR to update the original KEP section to include the feature gates. Hopefully it makes more sense now.
https://github.com/kubernetes/enhancements/pull/2966

Reply:
It's clear to me the flags all serve different purposes, but the combination of having to use three different feature flags doesn't make for a great user experience imho.

Wearing my PRR reviewer hat, I'm stuck because you're asking me to approve this changed based on the KEP-625 questionnaire, but you've added a new feature flag here that's not discussed there :) Hence, I can't do that because this third flag's behaviour isn't documented ... it needs to be added to the PRR questions.

Reply:
Let's continue discussion in #2966 which is adding documentation for the new feature gate.